### PR TITLE
UIDEXP-192 Rewrite Bigtest tests for duplicate mapping profile with jest/react-testing-library

### PR DESCRIPTION
**Pull request by UladzislauKutarkin**
## Purpose
https://issues.folio.org/browse/UIDEXP-192

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by vashjs** [comment]
Perhaps we should try to use the "msw" package to mock the server response, as we do in remote-storage. This would reduce the number of things we are currently mocking. 
This is a topic for discussion, maybe for now we can leave it that way and there will be other stories to add "msw". Who has any ideas?

**Review by vashjs — APPROVED** [review]

**Review by uladislausamets — APPROVED** [review]

File: src/settings/MappingProfiles/DuplicateMappingProfileRoute/DuplicateMappingProfileRoute.test.js [hunk 1]
```diff
@@ -90,37 +140,158 @@ describe('DuplicateMappingProfileRoute', () => {
...
+        userEvent.click(screen.getByRole('button', { name: 'Add transformations' }));
+        const transformationFields = getTransformationFieldGroups();
+        const modal = document.querySelector('[data-test-transformations-modal]');
+
+        userEvent.type(transformationFields[2].marcField.input, '');
+        userEvent.type(transformationFields[0].indicator2.input, '');
+        userEvent.type(transformationFields[0].subfield.input, '');
+
+        userEvent.click(screen.getByLabelText('Select all fields'));
+        userEvent.click(screen.getByLabelText('Select all fields'));
```
[the remark this answers is not in the document]
Comment (UladzislauKutarkin) [reply src/settings/MappingProfiles/DuplicateMappingProfileRoute/DuplicateMappingProfileRoute.test.js:220]: Changed to dbClick event. We need to double click check for keeping logic of the test case.

File: src/settings/MappingProfiles/DuplicateMappingProfileRoute/DuplicateMappingProfileRoute.test.js [hunk 2]
```diff
@@ -69,19 +73,65 @@ const DuplicateMappingProfileRouteContainer = ({
...
+describe('rendering duplicate mapping profile page without profile data', () => {
+  beforeEach(() => {
+    renderWithIntl(
+      <DuplicateMappingProfileRouteContainer profile={null} />,
+      translationsProperties
+    );
+  });
+
+  it('should display preloader', () => {
+    expect(screen.getByTestId('preloader')).toBeInTheDocument();
```
[the remark this answers is not in the document]
Comment (UladzislauKutarkin) [reply src/settings/MappingProfiles/DuplicateMappingProfileRoute/DuplicateMappingProfileRoute.test.js:85]: I agree, changing it now

File: src/settings/MappingProfiles/DuplicateMappingProfileRoute/DuplicateMappingProfileRoute.test.js [hunk 3]
```diff
@@ -69,19 +73,65 @@ const DuplicateMappingProfileRouteContainer = ({
...
 
+      expect(form).toBeInTheDocument();
+    });
+    it('should display correct pane title', () => {
+      const paneHeader = screen.getByRole('heading', { name: 'New field mapping profile' });
+
+      expect(getByText(paneHeader, translations['mappingProfiles.newProfile'])).toBeInTheDocument();
+    });
     it('should have correct name field value', () => {
       expect(screen.getByLabelText(/Name/)).toHaveValue(`Copy of ${mappingProfile.name}`);
```
[the remark this answers is not in the document]
Comment (UladzislauKutarkin) [reply src/settings/MappingProfiles/DuplicateMappingProfileRoute/DuplicateMappingProfileRoute.test.js]: Fixed.

**Review by axelhunn — APPROVED** [review]

**Review by vashjs — APPROVED** [review]

**Review by uladislausamets — APPROVED** [review]

**Pull request merged**